rmf_websocket/test/test_client_with_update_cb.cpp

```diff
@@ -103,9 +110,21 @@ TEST_CASE("Client", "Reconnecting server") {
         run_server();
       });
   t2.join();
-  REQUIRE(num_msgs == 2);
 
-  REQUIRE(msgs[0]["test"] == "1");
-  REQUIRE(msgs[1]["test"] == "2");
-  REQUIRE(num_init_msgs == 2);
+  // This is a horrible piece of code and defeats
+  // the purpose of the tests. But unfortunately websocketpp
+  // does not correctly return if a send was successful or not.
+  // Thus packets may be lost.
```

Comment:
For this test, we can use a promise to wait until the reconnect happens before sending the 2nd message. Something like

```cpp
const std::promise<void> on_init_promise;

auto init_function(std::promise<void>& promise) {
  ...
  promise.set_value();
}

REQUIRE_NO_THROW(on_init_promise.get_future().wait_for(1s));

// send 2nd message
```

Technically we can use this for `test_client.cpp` as well, but this requires the use of the reconnect init function, which kind of makes the 2 tests overlap.